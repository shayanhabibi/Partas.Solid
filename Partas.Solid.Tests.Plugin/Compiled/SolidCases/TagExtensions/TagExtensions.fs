module Partas.Solid.Tests.SolidCases.TagExtensions

open Partas.Solid
open Fable.Core

[<Erase>]
type NoPropsOneExtension() =
    interface RegularNode

    [<SolidTypeComponent>]
    member props.typeDef = div().attr ("key", "value")

[<Erase>]
type PropsWithExtension() =
    interface RegularNode

    [<SolidTypeComponent>]
    member props.typeDef = div(class' = "testClass").attr ("key", "value")

[<Erase>]
type NoPropsExtensionChain() =
    interface RegularNode

    [<SolidTypeComponent>]
    member props.typeDef = div().attr("key", "value").attr ("key2", "value2")

[<Erase>]
type PropsWithExtensionChain() =
    interface RegularNode

    [<SolidTypeComponent>]
    member props.typeDef =
        div(class' = "Tester", draggable = "drag").attr("key", "value").attr("key2", "value2").attr("key3", "value3").attr ("key4", "value4")
