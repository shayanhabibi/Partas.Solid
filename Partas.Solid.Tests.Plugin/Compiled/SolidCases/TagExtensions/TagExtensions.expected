import { splitProps } from "solid-js";

export function NoPropsOneExtension(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div key="value" />;
}

export function PropsWithExtension(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div class="testClass"
        key="value" />;
}

export function NoPropsExtensionChain(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div key="value"
        key2="value2" />;
}

export function PropsWithExtensionChain(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div class="Tester"
        draggable="drag"
        key="value"
        key2="value2"
        key3="value3"
        key4="value4" />;
}

